backend/lib/kubernetes-config/index.js

```diff
@@ -0,0 +1,277 @@
+    return { name, cluster }
+  }
+  const cleanContext = ({ name, context }) => {
+    context = _.pick(context, ['cluster', 'user', 'namespace'])
+    return { name, context }
+  }
+  const cleanAuthInfo = ({ name, user }) => {
+    user = _.pick(user, ['client-certificate-data', 'client-key-data', 'token', 'username', 'password'])
+    return { name, user }
+  }
+  const cleanConfig = ({
+    apiVersion = 'v1',
+    kind = 'Config',
+    clusters,
+    contexts,
+    'current-context': currentContext,
+    users
+  }) => {
+    return {
+      apiVersion,
+      kind,
+      clusters: _.map(clusters, cleanCluster),
+      contexts: _.map(contexts, cleanContext),
+      'current-context': currentContext,
+      users: _.map(users, cleanAuthInfo)
+    }
+  }
+  return cleanConfig(parseKubeconfig(input))
+}
+
+function fromKubeconfig (input) {
+  const {
+    clusters,
+    contexts,
+    'current-context': currentContext,
+    users
+  } = parseKubeconfig(input)
+
+  // inline certificates and keys
+  const readCertificate = (obj, name) => {
+    if (obj[name]) {
+      return fs.readFileSync(obj[name])
+    }
+    if (obj[`${name}-data`]) {
+      return Buffer.from(obj[`${name}-data`], 'base64').toString('utf8')
+    }
+  }
+
+  // get current user and cluster
+  const { context } = findByName(contexts, currentContext) || {}
+  const { cluster } = findByName(clusters, context.cluster) || {}
+  const { user } = findByName(users, context.user) || {}
+
+  const config = {
+    rejectUnauthorized: true
+  }
+
+  if (cluster) {
+    config.url = cluster.server
+    const ca = readCertificate(cluster, 'certificate-authority')
+    if (ca) {
+      config.ca = ca
+    }
+    if ('insecure-skip-tls-verify' in cluster) {
+      config.rejectUnauthorized = !cluster['insecure-skip-tls-verify']
+    }
+  }
+
+  if (user) {
+    const cert = readCertificate(user, 'client-certificate')
+    const key = readCertificate(user, 'client-key')
+    if (cert && key) {
+      config.cert = cert
+      config.key = key
+    }
+    if (user.token) {
+      config.auth = {
+        bearer: user.token
+      }
+    } else if (user.username && user.password) {
+      config.auth = {
+        user: user.username,
+        pass: user.password
+      }
+    }
+  }
+
+  return config
+}
+
+function dumpKubeconfig ({ user, context = 'default', cluster = 'garden', namespace, token, server, caData }) {
+  return yaml.safeDump({
+    apiVersion: 'v1',
+    kind: 'Config',
+    clusters: [{
+      name: cluster,
+      cluster: {
+        'certificate-authority-data': caData,
+        server
+      }
+    }],
+    users: [{
+      name: user,
+      user: {
+        token
+      }
+    }],
+    contexts: [{
+      name: context,
+      context: {
+        cluster,
+        user,
+        namespace
+      }
+    }],
+    'current-context': context
+  })
+}
+
+function mergeOptions (config, { auth, key, cert, privileged, ...options } = {}) {
```

Comment:
remove and implement only the two cases (auth + privileged) in createClient

Reply:
✔️